gem/gem.py

```diff
@@ -782,6 +783,58 @@ def __new__(cls, i, j):
         return self
 
 
+class Inverse(Node):
+    """ Inverse is a shape to shape operation.
+
+        :arg aggregate: Tensor to be inversed
+        :arg multiindex: Indices for result of Inverse
+
+        Inverse is different from ComponentTensor, because its children must have shape.
+        It does no transformation between free indices and shape, but the multiindex is still necessary.
+        The multiindices of the children are used for indexing them when passing to inverse function call,
+        the multiindex of the Inverse node itself is used to index the result of the function call.
+    """
+    __slots__ = ('children', 'shape')
+    __back__ = ()
+
+    def __new__(cls, aggregate):
+        assert aggregate.shape
+
+        self = super(Inverse, cls).__new__(cls)
+        self.children = (aggregate,)
+        self.shape = aggregate.shape
+        self.free_indices = aggregate.free_indices
+
+        return self
```

Comment:
You might as well just use `__init__` instead of `__new__`, since you never actually return any other type than an `Inverse`.  That spares you two lines: super-new and return-self.